dali/test/python/test_operator_readers_file.py

```diff
@@ -126,7 +129,65 @@ def test_file_reader_relpath_file_list():
             assert contents == ref_contents(fnames[index])
 
 
-batch_size_alias_test=64
+def _test_file_reader_filter(filters, batch_size, num_threads):
+    pipe = Pipeline(batch_size, num_threads, 0)
+    root = os.path.join(os.environ['DALI_EXTRA_PATH'], 'db/single/mixed')
+    files, labels = fn.readers.file(file_root=root, file_filters=filters)
+    pipe.set_outputs(files, labels)
+    pipe.build()
+    
+    fnames = set()
+    for label, dir in enumerate(sorted(next(os.walk(root))[1])):
+        for filter in filters:
+            for file in glob.glob(os.path.join(root, dir, filter)):
+                fnames.add((label, file.split('/')[-1], file))
+
+    fnames = sorted(fnames)
+
+    for i in range(len(fnames) // batch_size):
+        out_f, _ = pipe.run()
+        for j in range(batch_size):
+            with open(fnames[i * batch_size + j][2], 'rb') as file:
+                contents = np.array(list(file.read()))
+                assert all(contents == out_f.at(j))
+
+
+def _test_file_reader_filter_case_sensitive(filters):
```

Comment:
Can you deduplicate the code between both tests? The pipeline, file lookup and comparison are quite similar.